collector/task/ebpf/tcptop.go

```diff
@@ -0,0 +1,194 @@
+		singleton.enabled = enabled
+		singleton.ebpfAddr = ebpfAddr
+	})
+}
+
+// Metric contains values needed for planet metrics.
+type Metric struct {
+	Direction       string // ingress or egress
+	LocalHostgroup  string // e.g. hostgroup
+	RemoteHostgroup string
+	RemoteIPAddr    string
+	LocalDomain     string // e.g. consul domain
+	RemoteDomain    string
+	Bandwidth       float64
+}
+
+// Get returns latest metrics from singleton.
+func Get() []Metric {
+	singleton.mu.Lock()
+	hosts := singleton.hosts
+	singleton.mu.Unlock()
+
+	return hosts
+}
+
+// Collect will process ebpf metrics locally and fill singleton with latest data.
+func Collect(ctx context.Context) error {
+	if !singleton.enabled {
+		return nil
+	}
+
+	if singleton.ebpfAddr == "" {
+		return fmt.Errorf("eBPF address is empty")
+	}
+
+	startTime := time.Now()
+
+	// Scrape ebpf prometheus endpoint for send_bytes_metric and recv_bytes_metric.
+	ebpfScrape, err := prometheus.Scrape(singleton.ebpfAddr)
+	if err != nil {
+		return err
+	}
+	var send_bytes_metric *prom2json.Family
+	var recv_bytes_metric *prom2json.Family
+	for _, v := range ebpfScrape {
+		if v.Name == send_bytes {
+			send_bytes_metric = v
+		}
+		if v.Name == recv_bytes {
+			recv_bytes_metric = v
+		}
+		if send_bytes_metric != nil && recv_bytes_metric != nil {
+			break
+		}
+	}
+	if send_bytes_metric == nil {
+		return fmt.Errorf("Metric %v doesn't exist", send_bytes)
+	}
+	if recv_bytes_metric == nil {
+		return fmt.Errorf("Metric %v doesn't exist", recv_bytes)
+	}
+
+	sendHostBytes, err := toHostMetrics(send_bytes_metric, egress)
+	if err != nil {
+		log.Errorf("Conversion to host metric failed for %v, err: %v", send_bytes, err)
+	}
+	recvHostBytes, err := toHostMetrics(recv_bytes_metric, ingress)
+	if err != nil {
+		log.Errorf("Conversion to host metric failed for %v, err: %v", recv_bytes, err)
+	}
+
+	singleton.mu.Lock()
+	singleton.hosts = append(sendHostBytes, recvHostBytes...)
+	singleton.mu.Unlock()
+
+	log.Debugf("taskebpf.Collect retrieved %v metrics", len(sendHostBytes)+len(recvHostBytes))
+	log.Debugf("taskebpf.Collect process took %v", time.Since(startTime))
+	return nil
+}
+
+// toHostMetrics converts ebpf metrics into planet explorer prometheus metrics.
+func toHostMetrics(bytes_metric *prom2json.Family, direction string) ([]Metric, error) {
+	var hosts []Metric
+	inventoryHosts := inventory.Get()
+
+	localAddr, err := network.DefaultLocalAddr()
+	if err != nil {
+		return nil, err
+	}
+
+	// To label source traffic that we need to build dependency graph.
+	localHostgroup := localAddr.String()
+	localDomain := localAddr.String()
+	localInventory, ok := inventoryHosts[localAddr.String()]
+	if ok {
+		localHostgroup = localInventory.Hostgroup
+		localDomain = localInventory.Domain
+	}
+	log.Debugf("Local address doesn't exist in the inventory: %v", localAddr.String())
+
+	for _, m := range bytes_metric.Metrics {
+		metric := m.(prom2json.Metric)
+		destIp := net.ParseIP(metric.Labels["daddr"])
```

Comment:
Should we check for `"daddr"` key in metric.Labels and/or check whether [net.ParseIP](https://pkg.go.dev/net#ParseIP) returns non-nil value?
We may want to just `continue` in case that happens.

Reply:
Makes sense, added a nil check for destIp. 
Though in case of darkstat "ip" will almost always be there, but anyways good to have that check in. :)